src/NetTopologySuite/IO/GML2/GMLReader.cs

```diff
@@ -177,14 +185,68 @@ protected IEnumerable<Coordinate> ReadPosListAsCoordinates(int numOrdinates, str
             return coordinates;
         }
 
+        protected CoordinateSequence ReadPosList(XmlReader reader, int srsDimension, CoordinateSequenceFactory csFactory)
+        {
+            Assert.IsTrue(IsStartElement(reader, "posList"));
+
+            // Read, update srsDimension
+            srsDimension = ReadSrsDimension(reader, srsDimension);
+            reader.Read();
+
+            var coordinatesSpan = reader.ReadContentAsString().Trim().AsSpan();
+            var coordinates = new List<double>();
+
+            while (!coordinatesSpan.IsEmpty)
+            {
+                int seperatorIndex = coordinatesSpan.IndexOf(' ');
+                if (seperatorIndex == -1)
+                    seperatorIndex = coordinatesSpan.Length;
+
+                var ordinateSpan = coordinatesSpan.Slice(0, seperatorIndex);
+                if (!ordinateSpan.IsEmpty)
+                {
+#if NET7
+                    if (double.TryParse(ordinateSpan, System.Globalization.NumberStyles.Number,
+                        System.Globalization.NumberFormatInfo.InvariantInfo, out double ordinate))
```

Comment:
This part only does anything if we go back to multi-targeting.  Though it's not actually **wrong**, so I'm inclined to say let's keep it around.

Maybe `#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP2_1_OR_GREATER`, instead of `NET7`, since those are the minimum versions that we need for this API?